mongodb/configure.sls

```diff
@@ -0,0 +1,116 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+include:
+  - .service
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+
+place_mongodb_config_file:
+  file.managed:
+    - name: /etc/mongodb.conf
+    - template: jinja
+    - source: salt://mongodb/templates/mongodb.conf.j2
+    - require:
+      - file: copy_mongodb_key_file
+    - require_in:
+      - service: mongodb_service_running
+    - watch_in:
+      - service: mongodb_service_running
+
+wait_for_mongo:
+  cmd.run:
+    - name: |
+        until [ `netstat -tunlp | grep {{ mongodb.port }} | wc -l` -eq 1 ]
+        do
+          sleep 1
+        done
+    - shell: /bin/bash
+    - timeout: 60
+    - require:
+      - file: place_mongodb_config_file
+      - service: mongodb_service_running
+
+{% if 'mongodb_primary' in grains['roles'] %}
+
+{% if salt.pillar.get("mongodb:VAGRANT", false) %}
+{% do replset_config['members'].append({'_id': 0, 'host': '192.168.33.10:' + mongodb.port}) %}
+{% else %}
+{% set member_id = 0 %}
+{% for id, addrs in salt['mine.get']('roles:mongodb', 'network.get_hostname', expr_form='grain').items() %}
+{% do replset_config['members'].append({'_id': member_id, 'host': id}) %}
+{% set member_id = member_id + 1 %}
+{% endfor %}
+{% endif %}
+
+initiate_replset:
+  cmd.run:
+    - name: >
+        mongo --eval "printjson(rs.initiate({{ replset_config }}))"
+    - onlyif: "[ `mongo --eval 'printjson(rs.status())' | grep -i 'errmsg' | wc -l` -eq 1 ]"
+    - shell: /bin/bash
+    - require:
+        - cmd: wait_for_mongo
+
+wait_for_initialization:
+  cmd.run:
+    - name: |
+        until [ `mongo --eval 'printjson(rs.config())' | grep -i initializing | wc -l` -eq 0 ]
```

Comment:
We might want to use `-ge` instead of `-eq` in case it is listening on both IPv4 and IPv6 interfaces.